core/src/main/scala/zio/config/ConfigSourceModule.scala

```diff
@@ -378,12 +379,16 @@ trait ConfigSourceStringModule extends ConfigSourceModule {
       keyDelimiter: Option[Char],
       valueDelimiter: Option[Char],
       leafForSequence: LeafForSequence = LeafForSequence.Valid
-    ): IO[ReadError[String], ConfigSource] = {
+    ): ZIO[SystemModule, ReadError[String], ConfigSource] = {
```

Comment:
I think its a good idea, and gels well with ZIO's Environment.
 
Should we need another function along with this that return IO?, because I find a few users who like just using ZIO's IO bifunctor, probably by providing the system layer already, allowing them to do

```
for {
  env <- ConfigSource.fromSystemEnv
  prop <- ConfigSource.fromSystemProperties
  config. <- read(descriptor[AppConfig] from env.orElse(prop))
} yield config

```

Reply:
indeed - should be usefull, how should we name it then ?
Maybe: fromSystemEnvUnsafe ?
I need to think about proper name here then

Reply:
Ok Ive choosen: fromSystemEnvLive, hope its fine